lib/js/src/rpc/structs/VideoStreamingFormat.js

```diff
@@ -31,38 +31,54 @@
 */
 
 import { RpcStruct } from '../RpcStruct.js';
+import { VideoStreamingProtocol } from '../enums/VideoStreamingProtocol.js';
+import { VideoStreamingCodec } from '../enums/VideoStreamingCodec.js';
 
 class VideoStreamingFormat extends RpcStruct {
+    /**
+    * @constructor
+    */
+    constructor () {
+        super();
+    }
     /**
     * @param {VideoStreamingProtocol} val
     * @return {VideoStreamingFormat}
     */
     setProtocol (val) {
-        this.setParameter(VideoStreamingFormat.KEY_PROTOCOL, val);
-        return this;
+        const errorMessage = 'val must be an instance of VideoStreamingProtocol';
+        if (val instanceof VideoStreamingProtocol) {
+            this.setParameter(VideoStreamingFormat.KEY_PROTOCOL, val);
+            return this;
+        }
+        throw new Error(errorMessage);
     }
 
     /**
     * @return {VideoStreamingProtocol}
     */
     getProtocol () {
-        return this.getParameter(VideoStreamingFormat.KEY_PROTOCOL);
+        return this.getObject(VideoStreamingFormat.KEY_PROTOCOL);
     }
 
     /**
     * @param {VideoStreamingCodec} val
     * @return {VideoStreamingFormat}
     */
     setCodec (val) {
-        this.setParameter(VideoStreamingFormat.KEY_CODEC, val);
-        return this;
+        const errorMessage = 'val must be an instance of VideoStreamingCodec';
+        if (val instanceof VideoStreamingCodec) {
+            this.setParameter(VideoStreamingFormat.KEY_CODEC, val);
+            return this;
+        }
+        throw new Error(errorMessage);
     }
 
     /**
     * @return {VideoStreamingCodec}
     */
     getCodec () {
-        return this.getParameter(VideoStreamingFormat.KEY_CODEC);
+        return this.getObject(VideoStreamingFormat.KEY_CODEC);
```

Comment:
See previous comment about use of `getParameter` vs `getObject`

Reply:
fixing